engine/src/main/java/com/agiletec/plugins/jacms/aps/system/services/content/widget/ContentListHelper.java

```diff
@@ -252,23 +254,18 @@ protected boolean extractOrCategoryFilterClause(ApsProperties config) {
         return categoryCodes;
     }
 
-    @Deprecated
-    protected void addShowletFilters(IContentListTagBean bean, ApsProperties showletParams, String showletParamName, RequestContext reqCtx) {
-        this.addWidgetFilters(bean, showletParams, showletParamName, reqCtx);
-    }
-
-    protected void addWidgetFilters(IContentListTagBean bean, ApsProperties widgetParams, String widgetParamName, RequestContext reqCtx) {
+    protected EntitySearchFilter[] createWidgetFilters(IContentListTagBean bean, ApsProperties widgetParams, String widgetParamName, RequestContext reqCtx) {
         if (null == widgetParams) {
-            return;
+            return bean.getFilters();
         }
         String widgetFilters = widgetParams.getProperty(widgetParamName);
         EntitySearchFilter[] filters = this.getFilters(bean.getContentType(), widgetFilters, reqCtx);
         if (null == filters) {
-            return;
-        }
-        for (EntitySearchFilter filter : filters) {
-            bean.addFilter(filter);
+            return bean.getFilters();
         }
+        EntitySearchFilter[] filtersToReturn = (null != bean.getFilters()) ? Arrays.copyOf(bean.getFilters(), bean.getFilters().length) : new EntitySearchFilter[0];
```

Comment:
Way too complex for a trinary operator. Tough to see what is happening here. Please pul l this apart.

Reply:
Removed trinary operator